kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionEditor.vue

```diff
@@ -1,21 +1,401 @@
+              <KGrid>
+                <KGridItem
+                  :layout12="{ span: 1 }"
+                  :layout8="{ span: 1 }"
+                >
+                  <KIcon
+                    icon="dragVertical"
+                    class="space-content"
+                  />
+                </KGridItem>
+
+                <KGridItem
+                  :layout12="{ span: 6 }"
+                  :layout8="{ span: 5 }"
+                >
+                  <p class="space-content">
+                    {{ section.name.toUpperCase() }}
+                  </p>
+                </KGridItem>
+
+                <KGridItem
+                  :layout12="{ span: 5 }"
+                  :layout8="{ span: 2 }"
+                  class="current-section-style"
+                  :style="{ color: $themePalette.grey.v_700 }"
+                >
+                  <p class="current-section-text space-content">
+                    {{ section.current }}
+                  </p>
+                </KGridItem>
+              </KGrid>
+            </div>
+          </DragHandle>
+        </Draggable>
+      </transition-group>
+    </DragContainer>
+
+    <div class="bottom-buttons-style">
+      <KGrid>
+        <KGridItem
+          :layout12="{ span: 4 }"
+          :layout8="{ span: 2 }"
+          :layout4="{ span: 1 }"
+        >
+          <KButton
+            :text="deleteSection$()"
+            @click="quizForge.deleteSection(quizForge.activeSection.value.section_id)"
+          />
+        </KGridItem>
+        <KGridItem
+          :layout12="{ span: 8 }"
+          :layout8="{ span: 6 }"
+          :layout4="{ span: 3 }"
+        >
+          <KButton
+            :primary="true"
+            :text="applySettings$()"
+            class="apply-settings-style"
+            @click="quizForge.saveQuiz()"
+          />
+
+        </KGridItem>
+      </KGrid>
+    </div>
   </div>
 
 </template>
 
 
 <script>
 
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import useKResponsiveWindow from 'kolibri.coreVue.composables.useKResponsiveWindow';
+  import Draggable from 'kolibri.coreVue.components.Draggable';
+  import DragContainer from 'kolibri.coreVue.components.DragContainer';
+  import DragHandle from 'kolibri.coreVue.components.DragHandle';
+
   export default {
     name: 'SectionEditor',
+    components: {
+      Draggable,
+      DragContainer,
+      DragHandle,
+    },
+    inject: ['quizForge'],
+    setup() {
+      const {
+        sectionSettings$,
+        sectionTitle$,
+        numberOfQuestionsLabel$,
+        optionalDescriptionLabel$,
+        quizResourceSelection$,
+        numberOfSelectedResources$,
+        currentSection$,
+        deleteSection$,
+        applySettings$,
+        changeResources$,
+        sectionOrder$,
+        questionOrder$,
+        randomizedLabel$,
+        randomizedOptionDescription$,
+        fixedLabel$,
+        fixedOptionDescription$,
```

Comment:
Non-blocking thought here re: this new string convention (cc @marcellamaki @rtibbles )

I'm feeling of two minds on this convention where we destructure all of the message partials out and then return them in setup. With a lot of strings like this, it just adds 30 lines of cruft right at the top of the component definition.

So I'm wondering then if maybe we should, by convention, just return the strings object like:

```
setup() {
    /* ... */
    return {
         eqmStrings: enhancedQuizManagementStrings,
        /* ... */
    }
```

Then we'd use it like `this.eqmStrings.sectionLabel$()` or ` {{ eqmStrings.randomizedLabel$() }}`.

Just feels like the destructuring of the strings, as a pattern, would be better suited for when you're pulling one string out of the common core, but when you're pulling in a full feature's strings then you're liable to be using quite a lot of them.

In any case -- just want to get some thoughts.

Reply:
Sounds like a battle between:

> Explicit is better than implicit.

and

> Sparse is better than dense.

or

> Beautiful is better than ugly.


The laundry list doesn't particularly bother me, simply because of the explicitness involved, it tells us exactly which strings are going to be used in the component, and the component doesn't have to have any knowledge beyond the setup function, as to where those strings have come from.

Reply:
I was also wondering if a component that has a lot of strings will mean initializing the string from the setup and calling them the template or javascript will just increase the number of lines of code in the component.
I think your approach @nucleogenesis  would be good to limit that.